### PR TITLE
maintain state of recording starting in db for daily

### PR DESCRIPTION
**Pull request by zach-at-the-skills**

### CODE REVIEW

**Comment by zach-at-the-skills** [comment]
Only issue I really found is that sometimes the cloud recording fails unexpectedly, but our code is unaware of that and marks it as started regardless. There isn't really a great solution around that, though.

File: pages/call/[uid].tsx [hunk 1]
```diff
@@ -65,10 +65,27 @@ export default function JoinCall(props, session) {
...
-        showLeaveButton: true,
-      });
+      callFrame
+        .join({
+          url: props.booking.dailyRef.dailyurl,
+          showLeaveButton: true,
+        })
+        .then(() => {
+          if (!props.booking.dailyRef.recordingHasStarted) {
+            callFrame.startRecording({
```
Comment (mcasey8540) [inline pages/call/[uid].tsx]: @zach-at-the-skills - does `callFrame.startRecording` return a promise? if so, would be nice to put the fetch inside the `then` block.

File: pages/call/[uid].tsx [hunk 2]
```diff
@@ -100,15 +117,20 @@ export default function JoinCall(props, session) {
...
-          callFrame.startRecording({
-            width: 1280,
-            height: 720,
-            backgroundColor: "#FF1F2D3D",
-            layout: {
-              preset: "default",
-              max_cam_streams: 5,
-            },
-          });
+          if (!props.booking.dailyRef.recordingHasStarted) {
```
Comment (mcasey8540) [inline pages/call/[uid].tsx]: Opportunity to DRY up these similar code blocks

**Review by mcasey8540 — CHANGES_REQUESTED** [review]

File: pages/call/[uid].tsx (hunk 1 above)
Comment (zach-at-the-skills) [reply pages/call/[uid].tsx]: image
unfortunately it doesn't return a promise. No return (void).

**Review by mcasey8540 — APPROVED** [review]

**Pull request merged**